### PR TITLE
Data Apps Confirm modal

### PR DESCRIPTION
**Pull request by iethree**
## Description

Replaces the delete confirm modal, but this time, it can also be triggered by a custom action that has a `confirmMessage` set. (There is not currently any UI for users to set this, but I wanted to establish the codepath, so when we get a design for that, it should "just work" 🤞 )

Couple other minor changes:
- added some capitalization to modal titles
- added some special casing to the form submit buttons for `update` and `delete` implicit actions.
- deleted the old delete confirm modal (and some other code that depended on it, which is also deleted in #26062 )

Screen Shot 2022-10-21 at 4 19 07 PM

## Testing steps

- click a scaffolded delete button in a data app. you should get a delete confirm modal
- click delete and see that the record gets deleted 🔥

### CODE REVIEW

**Comment by kulyk** [comment]
@iethree some high-level visual feedback:

1. According to our [modal guidelines](https://www.notion.so/metabase/Modals-guidelines-fdf132ff081444088bcd37622fd8d023#28a633359c1e41b2a5a45edc1e2a4287), confirmation modal titles should always be questions (like "Delete?" / "Delete this record?" instead of just "Delete"), and the message should elaborate on the consequences. Think in our case, we should explain that it won't be possible to revert the delete.
2. Let's use the `"danger"` button variant for the "Delete" action
3. Let's add the "Cancel" button before "Delete", as we always show them explicitly in every confirmation modal

File: frontend/src/metabase/writeback/components/ActionCreator/FormCreator/utils.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import validate from "metabase/lib/validate";
 import type { Parameter } from "metabase-types/types/Parameter";
 import type { TemplateTag } from "metabase-types/types/Query";
 import type { Validator } from "metabase-types/forms";
+import { humanize } from "metabase/lib/formatting";
```
Comment (kulyk) [inline frontend/src/metabase/writeback/components/ActionCreator/FormCreator/utils.ts:15]: Think the `metabase/lib` import should come before `metabase-types` ?

**Review by kulyk — APPROVED** [review]
Looks and works good, but let's make the new modal match our design guidelines 👍

**Comment by iethree** [comment]
👍 Screen Shot 2022-10-24 at 11 14 14 AM

**Pull request merged**